system/ee/legacy/libraries/Stylesheet.php

```diff
@@ -25,135 +27,7 @@ class EE_Stylesheet
-
-        if ($stylesheet == '' or strpos($stylesheet, '/') === false) {
-            show_404();
-        }
-
-        if (! isset($this->style_cache[$stylesheet])) {
-            $ex = explode("/", $stylesheet);
-
-            if (count($ex) != 2) {
-                show_404();
-            }
-
-            ee()->db->select('templates.template_data, templates.template_name,	templates.edit_date');
-            ee()->db->from(array('templates', 'template_groups'));
-            ee()->db->where(
-                ee()->db->dbprefix('templates') . '.group_id',
-                ee()->db->dbprefix('template_groups') . '.group_id',
-                false
-            );
-            ee()->db->where('templates.template_name', $ex[1]);
-            ee()->db->where('template_groups.group_name', $ex[0]);
-            ee()->db->where('templates.template_type', 'css');
-            ee()->db->where('templates.site_id', ee()->config->item('site_id'));
-
-            $query = ee()->db->get();
-
-            if ($query->num_rows() == 0) {
-                show_404();
-            }
-
-            $row = $query->row_array();
-
-            /** -----------------------------------------
-            /**  Retreive template file if necessary
-            /** -----------------------------------------*/
-            if (ee()->config->item('save_tmpl_files') == 'y') {
-                ee()->load->helper('file');
-                $basepath = PATH_TMPL . ee()->config->item('site_short_name') . '/';
-                $basepath .= $ex['0'] . '.group/' . $row['template_name'] . '.css';
-
-                $str = read_file($basepath);
-                $row['template_data'] = ($str !== false) ? $str : $row['template_data'];
-                $row['edit_date'] = ($str !== false) ? filemtime($basepath) : $row['edit_date'];
-            }
-
-            $this->style_cache[$stylesheet] = str_replace(LD . 'site_url' . RD, stripslashes(ee()->config->item('site_url')), $row['template_data']);
-        }
-
-        $this->_send_css($this->style_cache[$stylesheet], $row['edit_date']);
-    }
-
-    /**
-     * EE Channel:form CSS
-     *
-     * Provides basic CSS for channel:form functionality on the frontend
-     *
-     * @return	void
-     */
-    private function _ee_channel_form_css()
-    {
-        $files[] = PATH_THEMES . 'cform/css/eecms-cform.min.css';
-
-        $out = '';
-
-        foreach ($files as $file) {
-            if (file_exists($file)) {
-                $out .= file_get_contents($file);
-            }
-        }
-
-        $out = str_replace('../../asset/', URL_THEMES_GLOBAL_ASSET, $out);
-
-        $this->_send_css($out, time());
-    }
-
-    /**
-     * Send CSS
-     *
-     * Sends CSS with cache headers
-     *
-     * @access	public
-     * @param	string	stylesheet contents
-     * @param	int		Unix timestamp (GMT/UTC) of last modification
-     * @return	void
-     */
-    public function _send_css($stylesheet, $modified)
```

Comment:
I'm concerned about removing this function because it is public. Removing this will break any addon currently using it, so I'd like to see us get it back in there.

Reply:
It's done, but I can't run tests:

```
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'Site_sites.site_color' in 'field list':
SELECT Site_sites.site_id as Site__site_id, Site_sites.site_label as Site__site_label, Site_sites.site_name as Site__site_name, Site_sites.site_description as Site__site_description, Site_sites.site_color as Site__site_color, Site_sites.site_bootstrap_checksums as Site__site_bootstrap_checksums, Site_sites.site_pages as Site__site_pages FROM (`exp_sites` as Site_sites) WHERE ( `Site_sites`.`site_id` IN (1) ) LIMIT 18446744073709551615
```

Reply:
It would be good to have this unit tested, is that something you would be able to add?

Reply:
Not yet. I'm a PHP newbie.

Reply:
All good! While it would be good to add unit tests, I'm alright with approving it for now since we're trying to get 6.3.0 ready to release and would like to get this in that release.

Reply:
So soon? Coooooooooool!